src/coreclr/src/vm/mlinfo.cpp

```diff
@@ -1595,55 +1594,52 @@ MarshalInfo::MarshalInfo(Module* pModule,
         }
 #endif // FEATURE_COMINTEROP
 
-        if (!fCalculatingFieldMetadata) // When calculating field metadata, we don't need to check the subtype of the pointer.
```

Comment:
If we want to fix https://github.com/dotnet/coreclr/issues/27800, instead of removing this condition and keeping the block, we should remove the block as a whole.

Reply:
I would do that in separate PR.